### ISSUE TITLE
Search user interface improvements

### ISSUE THREAD

**Issue by jtdaugherty** [matterhorn-chat/matterhorn#588]
@kquick mentioned this to me and can fill this out as desired, but what I remember is:
* `/search <terms>` is okay, but makes it difficult to know which special search terms influence the search (such as scoping the search to a specific channel)
* Making `/search` (no arguments) show an advanced search UI would be really nice; that way we can expose the various search controls and let the user modify them
* Showing the active search settings in search results would also be nice

This ticket needs research (API capabilities / search syntax) as well as design work (UI, UX) before we can move forward.

**Comment by jtdaugherty** [comment]
Due to extremely limited time on my part, I'm going to close this as something I am not going to be able to prioritize. If this needs to be prioritized in the future, let's re-open it or create a new ticket.
